api/v1beta1/clusterclass_types.go

```diff
@@ -637,6 +649,11 @@ type LocalObjectTemplate struct {
 	Ref *corev1.ObjectReference `json:"ref"`
 }
 
+// NamingStrategy defines a naming strategy for templated objects.
```

Comment:
We should probably use different structs so we can evolve it independently (e.g. when we want to add other names that don't exist for CP/MD/MP in the same way)

(but let's to this after we got consensus on the API)